### DISCUSSION TITLE
Non-exhaustive list checking

### DISCUSSION THREAD

**Discussion by oyarsa** [pydantic/pydantic#9542] [Question] [answered]
I'm using `TypeAdapter(list[T]).validate_json` to check if my data fits the correct shape. However, if N items in the list are missing a particular key (a common occurrence when the input has the wrong shape), Pydantic gives me N `missing` errors.

How can I make pydantic check only the first item in the list for validation?

**Comment by uriyyo** [comment]
Hi @oyarsa,

You can try to use generator_validator that works in a fail-fast way:
```py
from functools import partial
from typing import TYPE_CHECKING, TypeVar, TypeAlias, Annotated, Generator, Any

from pydantic import WrapValidator, TypeAdapter

if TYPE_CHECKING:
    T = TypeVar("T")

    FailFastList: TypeAlias = list[T]
else:
    T = TypeVar("T")


    def _fail_first_validator(gen_adapter: TypeAdapter, item: Any, _: Any) -> Any:
        return [*gen_adapter.validate_python(item)]


    class FailFastList:
        def __class_getitem__(cls, item):
            gen_adapter = TypeAdapter(Generator[item, Any, Any])

            return Annotated[
                list[T],
                WrapValidator(partial(_fail_first_validator, gen_adapter)),
            ]

TypeAdapter(FailFastList[int]).validate_json('["1", "2", "asdfsdf", {}, {}]')
```

In example above you will have only one validation error.

**Reply by oyarsa** [reply]
Thank you for the answer! I hoped there would be a setting to change this; your solution is quite verbose.

Would it be easier to emit just the first validation error instead? Pydantic could still check the entire list, but instead of reporting all N errors, it would report only the first one. Performance isn't really the issue here; it's more like the terminal output getting all the messages.

**Reply by uriyyo** [reply, marked as answer]
If you want just to trunc error, than it should be much easier:
```py
from typing import TypeVar, Annotated, Any

from pydantic import WrapValidator, ValidationError, TypeAdapter

T = TypeVar("T")


def _trunc_error(val: Any, next_: Any) -> Any:
    try:
        return next_(val)
    except ValidationError as e:
        raise ValidationError.from_exception_data(e.title, e.errors()[:1])


FirstErrorList = Annotated[
    list[T],
    WrapValidator(_trunc_error),
]

TypeAdapter(FirstErrorList[int]).validate_json('["1", "2", {}, "not-list"]')
```
